README.md

```diff
@@ -1,125 +1,175 @@
+Render the `ToastProvider` component in your app entry file.
 
 ```js
 // App.jsx
-import React from 'react';
-import Toast from 'react-native-toast-message';
+import React from 'react'
+import { ToastProvider } from '@stratuscode/react-native-toast-hook'
 
 function App(props) {
-  return (
-    <>
-      {/* ... */}
-      <Toast ref={(ref) => Toast.setRef(ref)} />
-    </>
-  );
+	return (
+		<>
+			<ToastProvider>
+				{/* ... */}
+			</ToastProvider>
+		</>
+	)
 }
 
-export default App;
+export default App
 ```
 
-Then use it anywhere in your app, by calling any `Toast` method directly:
+Then use it anywhere in your app with the `useToast` hook, which provides access to `showToast`, `queueToast`, `hideToast`, and `clearToastQueue`:
 
 ```js
-import Toast from 'react-native-toast-message';
-
-Toast.show({
-  text1: 'Hello',
-  text2: 'This is some something 👋'
-});
+import React from 'react'
+import { useToast } from '@stratuscode/react-native-toast-hook'
+
+function() {
+	const { queueToast } = useToast()
+
+	function makeToast() {
+		queueToast({
+			title: 'Hey you!',
+			message: 'The magic medicine works!',
+		})
+	}
+
+	return (
+		<View>
+			<Button onPress={makeToast} title="Toast It"/>
+		</View>
+	)
+}
 ```
 
-## API
-
-### `show(options = {})`
-
-When calling the `show` method, you can use the following `options` to suit your needs. Everything is optional, unless specified otherwise.
-
-The usage of `|` below, means that only one of the values show should be used.
-If only one value is shown, that's the default.
+If your toasts aren't showing up you might need to manually place the `Toaster`. For example, if you're using `react-navigation` you'll probably want the `Toaster` just before the end of your `</NavigationContainer>`. The `ToastProvider` component will render the `Toaster` component automatically unless you pass `renderToaster={false}` to the `ToastProvider`, so do that and then place your `Toaster` where you want it.
 
 ```js
-Toast.show({
-  type: 'success | error | info',
-  position: 'top | bottom',
-  text1: 'Hello',
-  text2: 'This is some something 👋',
-  visibilityTime: 4000,
-  autoHide: true,
-  topOffset: 30,
-  bottomOffset: 40,
-  onShow: () => {},
-  onHide: () => {}
-});
+import React from 'react'
+import { NavigationContainer } from '@react-navigation/native'
+import { createDrawerNavigator } from '@react-navigation/drawer'
+import { Toaster } from '@stratuscode/react-native-toast-hook'
+
+const Drawer = createDrawerNavigator()
+function Navigation() {
+	return (
+		<NavigationContainer>
+			<Drawer.Navigator {...etc} />
+			<Toaster />
+		</NavigationContainer>
+	)
+}
 ```
 
-### `hide(options = {})`
+## API
 
-```js
-Toast.hide({
-  onHide: () => {}
-});
+### `useToast()`
+
+The useToast hook gives you access to:
+- `queueToast: (toast: ToastProps) => void` - shows the toast immediately if one is not currently showing, otherwise queues up the toast to be shown after the current one goes away.
```

Comment:
bikeshed - ``appendToast`` instead?

Reply:
I still prefer "send/queue" over "send/append" as one is queueing it for a later showing. Append seems specific to how it's stored instead of how it's shown. Counterargument?

Reply:
What about getting rid of `queueToast` and adding a `priority` param to `showToast` that lets you choose between immediate, next, queue, or doNotQueue (never shows if it can't show immediately)? Default would be queue.